src/v/kafka/server/quota_manager.cc

```diff
@@ -72,6 +74,42 @@ quota_manager::maybe_add_and_retrieve_quota(
     return it;
 }
 
+void quota_manager::initialize_target_tp_rate_per_client_from_config() {
+    const std::vector<ss::sstring> clients_with_limit
+      = _target_tp_rate_per_client_group_config();
+    for (const auto& client_and_limit : clients_with_limit) {
+        auto parsing_setting = config::parse_client_group_byte_rate_quota(
+          client_and_limit);
+        _target_tp_rate_per_client_group.insert(parsing_setting.value());
+    }
+}
+
+std::optional<std::string_view> quota_manager::get_client_quota(
+  const std::optional<std::string_view>& client_id) {
+    if (!client_id) {
+        return std::nullopt;
+    }
+    for (const auto& group_and_limit : _target_tp_rate_per_client_group) {
+        if (client_id->starts_with(std::string_view(group_and_limit.first))) {
+            return group_and_limit.first;
+        }
+    }
+    return client_id;
+}
+
+int64_t quota_manager::get_client_target_tp_rate(
+  const std::optional<std::string_view>& quota_id) {
```

Comment:
Maybe rename all instances of quota_id to quota_group_id?

Reply:
It is not always group id.
If client has no group it is just client_id